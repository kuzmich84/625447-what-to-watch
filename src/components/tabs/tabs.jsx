import React from "react";
import {tabsTitle} from "../../const";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {fetchFilmReviews} from "../../store/api-actions";


const Tabs = (props) => {
  const {isActive, handlerTabOpen, filmId, loadReviewsServer} = props;
  return (
    <ul className="movie-nav__list">
      {tabsTitle.map((tab, i) => {
        return (
          <li
            key={i}
            className={`movie-nav__item  ${i === isActive ? `movie-nav__item--active` : ``} clicks-${i}`}
          >
            <a href="#" className="movie-nav__link" onClick={(e) => {
              e.preventDefault();
              handlerTabOpen(i);
              if (i === 2) {
                loadReviewsServer(filmId);
              }
            }}>{tab}</a>
          </li>
        );
      })}
    </ul>
  );
};

const mapDispatchToProps = (dispatch) => ({
  loadReviewsServer(filmId) {
    dispatch(fetchFilmReviews(filmId));
  }
});


export {Tabs};
export default connect(null, mapDispatchToProps)(Tabs);

Tabs.propTypes = {
  isActive: PropTypes.number.isRequired,
  handlerTabOpen: PropTypes.func.isRequired,
  filmId: PropTypes.number.isRequired,
  loadReviewsServer: PropTypes.func.isRequired,
};
