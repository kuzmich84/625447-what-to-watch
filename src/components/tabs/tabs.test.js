import React from "react";
import renderer from "react-test-renderer";
import {Tabs} from "./tabs";

const noop = () => {
};

test(`Should Tabs component render correctly`, () => {
  const tree = renderer
    .create(<Tabs
      handlerTabOpen={noop}
      isActive={1}
      filmId={10}
      loadReviewsServer={noop}
    />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

