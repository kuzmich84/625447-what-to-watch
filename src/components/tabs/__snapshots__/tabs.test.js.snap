// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs component render correctly 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item  "
  >
    <a
      className="movie-nav__link"
      href="#"
      onClick={[Function]}
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item  movie-nav__item--active"
  >
    <a
      className="movie-nav__link"
      href="#"
      onClick={[Function]}
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item  "
  >
    <a
      className="movie-nav__link"
      href="#"
      onClick={[Function]}
    >
      Reviews
    </a>
  </li>
</ul>
`;
